### PR TITLE
Release Refine

### LINKED ISSUE #251: Docker image releases with different Go version than binaries
```
$ oras version
Version:        0.11.1
Go version:     go1.16.2
Git commit:     5bfe0ab04a0f6fcdc5de4294cc79fb5b2e23a9a5
Git tree state: clean
```

```
$ docker run -it --rm -v $(pwd):/workspace ghcr.io/deislabs/oras:v0.11.1 version
Version:        0.11.1
Go version:     go1.15.7
Git commit:     5bfe0ab04a0f6fcdc5de4294cc79fb5b2e23a9a5
Git tree state: clean
```

How should we keep these best in sync other than using `latest` tag? Should we have a `GO_VERSION` file that is used both for CI and Dockerfile?

### LINKED ISSUE #364: Could it be possible to add 32 bits binaries?
Hi,

There is an amd64 and arm64 binaries availabe from the project, do you have the possibility to push the 32bits version for those artifacts?

### LINKED ISSUE #373: Multi-arch docker image
It would be great if you could publish a multi-architecture docker image that would include support for arm64.

### LINKED ISSUE #377: Refine release pipeline
This issue to track the enhancement on auto releasing `oras` when a new version is cut.

The current [release pipeline](https://oras.land/cli/5_developer_guide/#release-checklist) requires lots of manual steps and personal credentials. Instead of personal credentials, we can leverage [GITHUB_TOKEN](https://docs.github.com/en/actions/security-guides/automatic-token-authentication) for releases. An example can be found in the [notation](https://github.com/notaryproject/notation/blob/main/.github/workflows/release-github.yml) repo. Similarly, GITHUB_TOKEN can be used to release packages to `ghcr.io` according [this guide](https://docs.github.com/en/packages/managing-github-packages-using-github-actions-workflows/publishing-and-installing-a-package-with-github-actions#upgrading-a-workflow-that-accesses-ghcrio).

Before having a new release, it is better to revisit the following issues:
- #373
- #364 
- #251

Related issues:
- #256 

Successor issues:
- https://github.com/oras-project/oras-www/issues/10

### PR DESCRIPTION
**Pull request by qweeah**
This PR:
 - Automates releasing with `GITHUB_TOKEN`
 - Adds build for arm32 v7
 - Makes released docker image multi-arch

Once merged, will resolve #377, resolve #373, resolve #364, resolve #251

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 RUN apk add git make
 ENV ORASPKG /oras
 ADD . ${ORASPKG}
 WORKDIR ${ORASPKG}
-RUN make build-linux
-RUN mv ${ORASPKG}/bin/linux/amd64/oras /go/bin/oras
+RUN make "build-$(echo $TARGETPLATFORM | sed 's/\//-/g')"
+RUN mv ${ORASPKG}/bin/${TARGETPLATFORM}/oras /go/bin/oras
 
 FROM docker.io/library/alpine:3.13.5
```
Comment (shizhMSFT) [inline Dockerfile:10]: Do we want to upgrade the alpine version like `3.15.4`?

File: .github/workflows/release-github.yml [hunk 2]
```diff
@@ -21,4 +21,4 @@ jobs:
           version: latest
           args: release --rm-dist
         env:
-          GITHUB_TOKEN: ${{ secrets.RELEASE_GITHUB_USER_TOKEN }}
```
[the remark this answers is not in the document]
Comment (shizhMSFT) [reply .github/workflows/release-github.yml:24]: similar for other yaml files

**Comment by shizhMSFT** [comment]
Have you also checked `.goreleaser.yml`?

**Comment by qweeah** [comment]
> Have you also checked `.goreleaser.yml`?

I have added arm32 builds for go releaser.See in the latest change.

File: Makefile [hunk 3]
```diff
@@ -30,10 +30,10 @@ clean:
 	git status --ignored --short | grep '^!! ' | sed 's/!! //' | xargs rm -rf
 
 .PHONY: build
-build: build-linux build-linux-arm64 build-mac build-mac-arm64 build-windows
+build: build-linux-amd64 build-linux-arm64 build-mac build-mac-arm64 build-windows
 
-.PHONY: build-linux
-build-linux:
+.PHONY: build-linux-amd64
```
[the remark this answers is not in the document]
Comment (qweeah) [reply Makefile:38]: Resolve as suggested.

File: .github/workflows/release-github.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (qweeah) [reply .github/workflows/release-github.yml:24]: Updated all the updatedables.

File: Dockerfile (hunk 1 above)
Comment (qweeah) [reply Dockerfile]: Upgraded to the latest.

File: .github/workflows/release-ghcr.yml [hunk 4]
```diff
@@ -21,18 +21,16 @@ jobs:
...
-          username: ${{ secrets.RELEASE_GHCR_USER_NAME }}
-          password: ${{ secrets.RELEASE_GHCR_USER_TOKEN }}
-      - name: docker push
-        run: |
-          docker push ${{ steps.prepare.outputs.ref }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: docker build
+        run: |          
+          docker buildx create --use --name build --node build --driver-opt network=host
```
[the remark this answers is not in the document]
Comment (qweeah) [reply .github/workflows/release-ghcr.yml]: Removed

File: Makefile [hunk 5]
```diff
@@ -48,6 +51,9 @@ build-linux-arm-v7:
 		-o bin/linux/arm/v7/$(CLI_EXE) $(CLI_PKG)
 
 .PHONY: build-mac
+build-mac: build-mac-arm64 build-mac-amd64
+
+.PHONY: build-mac-amd64
 build-mac:
```
Comment (shizhMSFT) [inline Makefile:57]:
```suggestion
build-mac-amd64:
```

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**